### ISSUE TITLE
SHA check fails

### ISSUE THREAD

**Issue by mbr0wn** [gnuradio/gnuradio-livesdr#16]
My brain is fried and tomorrow this will probably be obvious. But the verification always fails (regardless of the state of my repo). Do I need import a key? Anyway, the instructions don't seem to be complete. I'm running on 16.04 btw.

```
$ make
/home/mbant/src/gnuradio-livesdr/iso/ubuntu-14.04.5-desktop-amd64.iso
[INFO] Downloading stock Ubuntu ISO image if needed...
[INFO] Not found, downloading http://releases.ubuntu.com/14.04/ubuntu-14.04.5-desktop-amd64.iso
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1052M  100 1052M    0     0  4381k      0  0:04:06  0:04:06 --:--:-- 5329k
[INFO] Checking for Ubuntu gpg public key (0xFBB75451)
[INFO] Retrieving SHA256 hashes and signature files
######################################################################## 100.0%
######################################################################## 100.0%
gpg: Signature made Thu 04 Aug 2016 01:46:28 PM PDT using DSA key ID FBB75451
gpg: Good signature from "Ubuntu CD Image Automatic Signing Key <cdimage@ubuntu.com>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: C598 6B4F 1257 FFA8 6632  CBA7 4618 1433 FBB7 5451
gpg: Signature made Thu 04 Aug 2016 01:46:28 PM PDT using RSA key ID EFE21092
gpg: Can't check signature: public key not found
[ERROR] SHA256SUMS signature verification failed!
Makefile:28: recipe for target '~/src/gnuradio-livesdr/iso/ubuntu-14.04.5-desktop-amd64.iso' failed
make: *** [~/gnuradio-livesdr/iso/ubuntu-14.04.5-desktop-amd64.iso] Error 1
```

**Comment by jmcorgan** [comment]
Yeah, I do think there is a pubkey you're missing, and it might need to get put into the make deps target. This is an example of something you'd PR against the home repository master branch on jmcorgan repo.

**Comment by mbr0wn** [comment]
Hm, or not. I just manually downloaded the ISO and put it into the iso/ directory. Then it worked. It seems like the `curl` command left a `*.iso.tmp` file in the iso dir, instead of a .iso file.

**Comment by jmcorgan** [comment]
The script should rename the .tmp file on a successful download to the .iso file for the check. So what's probably happening is broken logic in the script for dealing with interrupted transfers.

**Comment by jmcorgan** [comment]
More accurately, the curl command gives the download a .iso.tmp file name, and that is what the signature is checked against.  Once that succeeds, then the file is renamed to .iso.

In your case, by renaming the tmp file to the iso file, the makefile assumes the stock ISO is in place and goes on to the next step.

I still think the get-stock-iso script is not robust to failed or interrupted transfers, then resuming another make invocation to continue.

**Comment by jmcorgan** [comment]
So, there's a bug here somewhere, but if you have the ISO in place you can continue working normally.

**Comment by mbr0wn** [comment]
The .tmp file had the right size and sha256. I think I might have been successful just renaming the .tmp to .iso.

**Comment by mbr0wn** [comment]
Sorry misread your previous comment. OK, so the GPG check is failing. But the sha256sum was the right one.

**Comment by jmcorgan** [comment]
This was fixed in 06d2abb.
